runtime/dart_isolate.cc

```diff
@@ -678,7 +686,12 @@ fxl::RefPtr<DartSnapshot> DartIsolate::GetIsolateSnapshot() const {
 }
 
 fml::WeakPtr<DartIsolate> DartIsolate::GetWeakIsolatePtr() const {
-  return weak_factory_.GetWeakPtr();
+  return (weak_factory_) ? weak_factory_->GetWeakPtr()
+                         : fml::WeakPtr<DartIsolate>();
+}
+
+void DartIsolate::ResetWeakPtrFactory() {
+  weak_factory_.reset();
```

Comment:
Check if a weak factory exists here too. Just being paranoid in case there are more callers in the future.